tests/test_bootloader.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+import os
+import subprocess
+from tempfile import NamedTemporaryFile, TemporaryFile
+
+from xcp.bootloader import Bootloader
+
+class TestBootloader(unittest.TestCase):
+    def test_grub2(self):
+        bl = Bootloader.readGrub2("tests/data/grub.cfg")
+        with NamedTemporaryFile("w") as temp:
+            bl.writeGrub2(temp.name)
+            # get a diff
+            proc = subprocess.Popen(["diff", "tests/data/grub.cfg", temp.name],
+                                    stdout = subprocess.PIPE,
+                                    universal_newlines=True)
+        for line in proc.stdout:
+            # FIXME: check is entirely ad-hoc, should we have a diff at all ?
+            self.assertRegexpMatches(line, r"^(5a6,13$|>)")
+
+
+class TestBootloaderAdHoc(unittest.TestCase):
```

Comment:
I wonder if grub1 and extlinux are actually used nowadays.

Reply:
I too, but the code is still there

Reply:
Grub1 maybe not.  extlinux, yes :-(